api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java

```diff
@@ -105,7 +106,8 @@
     public ResponseEntity<Long> countParticipants(Long cdrVersionId, SearchRequest request) {
         CdrVersionContext.setCdrVersion(cdrVersionDao.findOne(cdrVersionId));
 
-        QueryJobConfiguration qjc = bigQueryService.filterBigQueryConfig(participantCounter.buildParticipantCounterQuery(request));
+        QueryJobConfiguration qjc = bigQueryService.filterBigQueryConfig(participantCounter.buildParticipantCounterQuery(
+            new ParticipantCriteria(request)));
```

Comment:
This feels a little clunky to me as it is really not clear why I or another developer would need to wrap search request for the cohort builder with a ParticipantCriteria. Would it make more sense to push this into the ParticipantCounter or to use an interface as the incoming parameter?

Reply:
My thought is that SearchRequest currently serves to identify which participants you want to retrieve data on, but doesn't allow you to whitelist / blacklist participant IDs... wrapping SearchRequest in ParticipantCriteria allows for that.  In future we might have other ways of indicating what participants we want to pull out, which could also be put into ParticipantCriteria.

We could limit it just to the methods used by the endpoint for materializing cohorts, but it seemed a bit more flexible / cleaner to just switch everything over. Let me know if you disagree.

I could make ParticipantCriteria an interface rather than a class, but either way you'd have to construct something that wraps SearchRequest... so I'm not sure it helps?

(One other option I briefly considered: specifying participant ID whitelists and blacklists using SearchRequest children. It seemed like the SQL generated from that was going to be uglier than doing things this way, and I wasn't sure we really wanted clients to be able to pass us participant ID lists in their cohort criteria, so I decided not to do this.)

Reply:
If regarding the interface thing, you meant that we could have an interface like ParticipantCriteria with implementing classes SearchRequestAndParticipantBlacklistCriteria and ParticipantWhitelistCriteria, each which knew how to make their own SQL statements for selecting participants.... I could imagine doing that. Just seemed a little bit more complicated, and spreads out where SQL gets generated more.

Reply:
True, the interface approach would spread out the implementation. I can go either way on this, but if we go the ParticipantCriteria approach maybe just put some comments explaining your thoughts about how to use this class like above.

Reply:
Done.